fuzzy-native/src/MatcherBase.cpp

```diff
@@ -35,12 +35,51 @@ inline string str_to_lower(const std::string &s) {
   return lower;
 }
 
+float distance_to_root(const std::string *root, const std::string *value) {
+  size_t num_common_dirs = 0;
+  size_t i = 0;
+
+  // Count number of common directories
+  for (; i < root->length() && i < value->length(); i++) {
+    if (root->at(i) != value->at(i)) {
+      break;
+    }
+    if (root->at(i) == '/' || root->at(i) == '\\') {
+      num_common_dirs++;
+    }
+  }
+
+  if (i == root->length()) {
+    return 0;
+  }
+
+  // Count total number of directories in root
```

Comment:
Since we're using the same root throughout, this should be pre-computed and passed around in `MatchOptions`.